src/hooks/users.ts

```diff
@@ -1,75 +1,53 @@
-import { useMemo } from "react";
+import { useEffect } from "react";
+import { shallowEqual } from "react-redux";
+import { isEqual } from "lodash";
 
-import { User } from "types/User";
+import { User, UserLocation, userWithLocationToUser } from "types/User";
+import { ReactHook } from "types/utility";
 
-import { WithId } from "utils/id";
+import { withId, WithId } from "utils/id";
 import { normalizeTimestampToMilliseconds } from "utils/time";
 
-import { worldUsersByIdSelector, worldUsersSelector } from "utils/selectors";
+import {
+  worldUsersByIdSelector,
+  worldUsersByIdWithoutLocationSelector,
+  worldUsersSelector,
+} from "utils/selectors";
+import { useWorldUsersContext } from "./users/useWorldUsers";
 
 import { useSelector } from "./useSelector";
 import { useUserLastSeenThreshold } from "./useUserLastSeenThreshold";
-import { useConnectCurrentVenueNG } from "./useConnectCurrentVenueNG";
-import { useVenueId } from "./useVenueId";
-import { useFirestoreConnect, isLoaded } from "./useFirestoreConnect";
-import { useSovereignVenueId } from "./useSovereignVenueId";
+import { isLoaded } from "./useFirestoreConnect";
 
-export const useConnectWorldUsers = () => {
-  const venueId = useVenueId();
-
-  const { sovereignVenueId, isSovereignVenueIdLoading } = useSovereignVenueId({
-    venueId,
-  });
-
-  useFirestoreConnect(() => {
-    if (isSovereignVenueIdLoading || !sovereignVenueId || !venueId) return [];
-
-    const relatedLocationIds = [venueId];
-
-    if (sovereignVenueId) {
-      relatedLocationIds.push(sovereignVenueId);
-    }
-
-    return [
-      {
-        collection: "users",
-        where: ["enteredVenueIds", "array-contains-any", relatedLocationIds],
-        storeAs: "worldUsers",
-      },
-    ];
-  });
-};
-
-export const useWorldUsers = (): {
-  worldUsers: readonly WithId<User>[];
-  isWorldUsersLoaded: boolean;
-} => {
-  useConnectWorldUsers();
+export { useWorldUsers } from "./users/useWorldUsers";
 
-  const selectedWorldUsers = useSelector(worldUsersSelector);
+const noUsers: WithId<User>[] = [];
+const noUsersById: Record<string, WithId<User>> = {};
```

Comment:
> Can we start adding these in a more global/reusable file instead?

If we did that, then they would have the same reference for potentially unrelated things. Not sure if that would matter in reality or not.. but it might. Aside from that edge case, I don't have any strong feels against moving them to a common location.

> How about we use closures instead, that way it will be initialized only when the function is called.

What do you mean by this?

Reply:
>>How about we use closures instead, that way it will be initialized only when the function is called.

>What do you mean by this?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures
https://www.w3schools.com/js/js_function_closures.asp

Maybe it's better to initialize the value only when it's needed (in the function itself), otherwise it will be always created globally, even if that function is not called and that value is not actually used or needed.

Reply:
If this was for some super heavyweight thing I might agree with using a lazy initialisation pattern. But IMO in doing so we're likely going to create excess bloat and complicate what the compiler/browser may just optimise out for us anyway. I don't think it's worth spending the time thinking about this tbh.